platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/status/PlatformStatusManager.java

```diff
@@ -0,0 +1,140 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.system.status;
+
+import com.swirlds.base.state.Startable;
+import com.swirlds.base.state.Stoppable;
+import com.swirlds.base.time.Time;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.notification.NotificationEngine;
+import com.swirlds.common.system.status.actions.PlatformStatusAction;
+import com.swirlds.common.system.status.actions.TimeElapsedAction;
+import com.swirlds.common.threading.framework.QueueThread;
+import com.swirlds.common.threading.framework.config.QueueThreadConfiguration;
+import com.swirlds.common.threading.framework.config.QueueThreadMetricsConfiguration;
+import com.swirlds.common.threading.manager.ThreadManager;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+
+/**
+ * An asynchronous component which manages platform status.
+ * <p>
+ * This object wraps a {@link PlatformStatusStateMachine}, which contains the actual state machine logic.
+ */
+public class PlatformStatusManager implements PlatformStatusGetter, StatusActionSubmitter, Startable, Stoppable {
+    /**
+     * A source of time
+     */
+    private final Time time;
+
+    /**
+     * The queue to handle incoming {@link PlatformStatusAction}s
+     */
+    private final QueueThread<PlatformStatusAction> queue;
+
+    /**
+     * The platform status state machine that is being wrapped by this asynchronous component
+     */
+    private final PlatformStatusStateMachine stateMachine;
+
+    /**
+     * Constructor
+     *
+     * @param platformContext    the platform context
+     * @param time               a source of time
+     * @param threadManager      the thread manager
+     * @param notificationEngine the notification engine
+     */
+    public PlatformStatusManager(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final Time time,
+            @NonNull final ThreadManager threadManager,
+            @NonNull final NotificationEngine notificationEngine) {
+
+        this.time = Objects.requireNonNull(time);
+        Objects.requireNonNull(threadManager);
+        Objects.requireNonNull(notificationEngine);
+
+        final PlatformStatusConfig config =
+                platformContext.getConfiguration().getConfigData(PlatformStatusConfig.class);
+        this.stateMachine = new PlatformStatusStateMachine(time, config, notificationEngine);
+
+        this.queue = new QueueThreadConfiguration<PlatformStatusAction>(threadManager)
+                .setComponent("platform-status")
+                .setThreadName("platform-status-state-machine")
```

Comment:
Optional suggestion.
```suggestion
                .setComponent("platform")
                .setThreadName(status-state-machine")
```

This will cause the thread name to be `<platform: status-state-machine>`